workspace/__lib__/xod/icons/mask-color/patch.cpp

```diff
@@ -0,0 +1,16 @@
+
+struct State {
+};
+
+{{ GENERATED_CODE }}
+
+void evaluate(Context ctx) {
+    auto bitmap = getValue<input_IN>(ctx);
+
+    if (bitmap->colorDepth > 0) {
+        bitmap->alphaColor = getValue<input_MSK>(ctx);
+        bitmap->colorDepth = 2;
+    }
```

Comment:
Upstream object mutation is quite a bad idea here. Imagine two `mask-color` share the same input bitmap: only one will “win”. Worse, the side-effect is hidden. The node has no input pulses to do the destructive action. I see two possible solutions:

1. Return a *copy* of the bitmap which shares the same pix buffer but with `alphaColor` and `colorDepth` fields updated.
2. Do not introduce `mask-color` at all: keep the whole image and format definition in the bitmap constructor nodes